ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroDeserializerTest.java

```diff
@@ -53,236 +62,733 @@
-                                              final Object avroValue,
-                                              final Schema ksqlSchema) {
-    shouldDeserializeTypeCorrectly(avroSchema, avroValue, ksqlSchema, avroValue);
+  @Test
+  public void shouldDeserializedAvroBoolean() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_BOOLEAN_SCHEMA);
+
+    final byte[] bytes = givenAvroSerialized(true, BOOLEAN_AVRO_SCHEMA);
+
+    // When:
+    final Object result = deserializer.deserialize(SOME_TOPIC, bytes);
+
+    // Then:
+    assertThat(result, is(true));
   }
 
-  private void shouldDeserializeTypeCorrectly(final org.apache.avro.Schema avroSchema,
-                                              final Object avroValue,
-                                              final Schema ksqlSchema,
-                                              final Object ksqlValue) {
+  @Test
+  public void shouldThrowIfCanNotCoerceToBoolean() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_BOOLEAN_SCHEMA);
 
+    final byte[] bytes = givenAvroSerialized(10, LONG_AVRO_SCHEMA);
 
-    final org.apache.avro.Schema avroRecordSchema = org.apache.avro.SchemaBuilder.record("test_row")
-        .fields()
-        .name("field0")
-        .type(avroSchema)
-        .noDefault()
-        .endRecord();
-    final Schema ksqlRecordSchema = SchemaBuilder.struct().field("field0", ksqlSchema).build();
+    // Then:
+    expectedException.expect(SerializationException.class);
+    expectedException.expectCause(hasMessage(containsString(
+        "Cannot deserialize type int32 as type boolean")));
 
-    final GenericRecord avroRecord = new GenericData.Record(avroRecordSchema);
-    avroRecord.put("field0", avroValue);
+    // When:
+    deserializer.deserialize(SOME_TOPIC, bytes);
+  }
 
-    final Struct row = serializeDeserializeAvroRecord(
-        ksqlRecordSchema, "test-topic", schemaRegistryClient, avroRecord);
+  @Test
+  public void shouldDeserializedAvroInt() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_INT32_SCHEMA);
 
-    assertThat(row.schema(), is(ksqlRecordSchema));
-    assertThat(row.get("field0"), equalTo(ksqlValue));
+    final byte[] bytes = givenAvroSerialized(42, INT_AVRO_SCHEMA);
+
+    // When:
+    final Object result = deserializer.deserialize(SOME_TOPIC, bytes);
+
+    // Then:
+    assertThat(result, is(42));
+  }
+
+  @Test
+  public void shouldThrowIfCanNotCoerceToInt() {
+    // Given:
+    givenDeserializerForSchema(Schema.OPTIONAL_INT32_SCHEMA);
+
+    final byte[] bytes = givenAvroSerialized(true, BOOLEAN_AVRO_SCHEMA);
+
+    // Then:
+    expectedException.expect(SerializationException.class);
+    expectedException.expectCause(hasMessage(containsString(
+        "Cannot deserialize type boolean as type int32")));
+
+    // When:
+    deserializer.deserialize(SOME_TOPIC, bytes);
+  }
+
+  @Test
+  public void shouldDeserializedAvroBigInt() {
```

Comment:
We already have test cases for these (for e.g. see lines 817, 1062, 1072, etc) - let's consolidate. Ditto for deserializing into int

Reply:
Ah, difference is that the preexisting tests deserialize a BigInt _field_, right!